pootle/middleware/auth.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+"""
+Custom Pootle authentication middleware that takes care of returning Pootle's
+special `nobody` user instead of Django's own `AnonymousUser`.
+
+The code has been slightly adapted from django.contrib.auth.middleware.
+
+Note this customization would probably be unnecessary if there was a fix for
+https://code.djangoproject.com/ticket/20313
```

Comment:
Thanks for the pointer @unho. In fact, I ended up removing the `User.get` method as it has a single and very specific use which is fine to do directly in the middleware, as opposed to the previous scenario where it was necessary to do it manually in view code. So fortunately there's no need to be concerned about it anymore :smile: